app/controllers/Application.scala

```diff
@@ -60,73 +68,48 @@ class Application(
         contributionsPayPalEndpoint,
         description = Some(stringsConfig.bundleLandingDescription)
       ))
-    } else if (newDesigns == "circles-garnett") {
+    } else {
+      val (updatedId, updatedJs) = applyCircles(newDesigns, id, js, "support-landing-page", "supportLandingPage.js")
       Ok(views.html.bundleLanding(
         title,
-        "support-landing-page",
-        "supportLandingPage.js",
+        updatedId,
+        updatedJs,
         contributionsPayPalEndpoint,
         description = Some(stringsConfig.bundleLandingDescription)
       ))
-    } else {
-      Ok(views.html.bundleLanding(title, id, js, contributionsPayPalEndpoint, description = Some(stringsConfig.bundleLandingDescription)))
     }
   }
 
   def regularContributionsThankYou(title: String, id: String, js: String, newDesigns: String): Action[AnyContent] = CachedAction() { implicit request =>
-    if (newDesigns == "circles-garnett") {
-      Ok(views.html.react(title, "regular-contributions-thank-you-page", "regularContributionsThankYouPage.js"))
-    } else {
-      Ok(views.html.react(title, id, js))
-    }
+    val (updatedId, updatedJs) = applyCircles(newDesigns, id, js, "regular-contributions-thank-you-page", "regularContributionsThankYouPage.js")
+    Ok(views.html.react(title, updatedId, updatedJs))
   }
 
   def contributionsLandingUK(title: String, id: String, js: String, newDesigns: String): Action[AnyContent] = CachedAction() { implicit request =>
-    if (newDesigns == "circles-garnett") {
-      Ok(views.html.contributionsLanding(
-        title,
-        description = Some(stringsConfig.contributionLandingDescription),
-        "contributions-landing-page-uk",
-        "contributionsLandingPageUK.js",
-        contributionsPayPalEndpoint
-      ))
-    } else {
-      Ok(views.html.contributionsLanding(
-        title,
-        description = Some(stringsConfig.contributionLandingDescription),
-        id,
-        js,
-        contributionsPayPalEndpoint
-      ))
-    }
+    val (updatedId, updatedJs) = applyCircles(newDesigns, id, js, "contributions-landing-page-uk", "contributionsLandingPageUK.js")
```

Comment:
Thank you. This is definitely cleaner. There is still room for improvement if you wish: both `contributionsLandingUK` and `contributionsLandingUs` are almost the same except for the id and js suffixes.